lib/data/dataloader.js

```diff
@@ -361,8 +361,10 @@ function loadFood(ddata, ctx, callback) {
 }
 
 function loadDeviceStatus(ddata, env, ctx, callback) {
+    var retroDays = 1;
+    if(env.extendedSettings.devicestatus && env.extendedSettings.devicestatus.days) retroDays = env.extendedSettings.devicestatus.days;
     var dateRange = {
-        $gte: new Date(ddata.lastUpdated - ONE_DAY).toISOString()
+        $gte: new Date( ddata.lastUpdated - (ONE_DAY * retroDays) ).toISOString()
```

Comment:
I think this would be much cleaner if instead of multiplying the constant you'd add a new TWO_DAYS constant and assign a variable to the constant based on the config

Reply:
I think there may be a TWO_DAYS constant in that file already - if so I'll use that. I had originally been intending to use a multiplier but decided to limit it to two days for the reasons you mentioned on performance.